pkg/test/extensions/binary.go

```diff
@@ -457,3 +476,14 @@ func runWithTimeout(ctx context.Context, c *exec.Cmd, timeout time.Duration) ([]
 	}
 	return c.CombinedOutput()
 }
+
+// safeComponentPath sanitizes a component identifier to be safe for use as a file or directory name.
+func safeComponentPath(c *Component) string {
+	safeRegexp := regexp.MustCompile(`[<>:"/\\|?*\s]+`)
```

Comment:
could be defined as a var outside the function for re-use without recompilation

Reply:
I did update it, but the savings here is extremely marginal.

About 750ns per invocation (2.5ms per 3K test runs -- i.e. one job run).  Computing the artifact once saves a little more, about 9ms per run, which is what I did.

```
goos: darwin
goarch: arm64
pkg: example.com/bench
cpu: Apple M3 Max
BenchmarkSafeComponentPathLocal-14     	  667196	      1782 ns/op
BenchmarkSafeComponentPathGlobal-14    	 1000000	      1052 ns/op
```